evalml/pipelines/components/transformers/preprocessing/time_series_featurizer.py

```diff
@@ -160,62 +170,115 @@ def _find_significant_lags(y, conf_level, max_delay):
             significant_lags = all_lags
         return significant_lags
 
-    def transform(self, X, y=None):
-        """Computes the delayed features for all features in X and y.
+    def _compute_rolling_transforms(self, X, y, original_features):
+        """Compute the rolling features from the original features.
 
-        For each feature in X, it will add a column to the output dataframe for each
-        delay in the (inclusive) range [1, max_delay]. The values of each delayed feature are simply the original
-        feature shifted forward in time by the delay amount. For example, a delay of 3 units means that the feature
-        value at row n will be taken from the n-3rd row of that feature
+        Args:
+            X (pd.DataFrame or None): Data to transform.
+            y (pd.Series, or None): Target.
 
-        If y is not None, it will also compute the delayed values for the target variable.
+        Returns:
+            pd.DataFrame: Data with rolling features. All new features.
+        """
+        size = int(self.rolling_window_size * self.max_delay)
+        rolling_mean = RollingMean(
+            window_length=size + 1,
+            gap=self.start_delay,
+            min_periods=size + 1,
+        )
+        rolling_mean = rolling_mean.get_function()
```

Comment:
I tried computing the features with dfs but IMO it's a bit awkward to have one set of features go through featuretools while the other does not: https://github.com/alteryx/evalml/pull/3088/

I spoke with the featuretools team, and it's probably best to wait until they release Lagged rolling primitives so we can refactor the whole component to use dfs at that point.